src/test/user.spec.js

```diff
@@ -192,3 +200,139 @@ describe('Test user login, signup and account verification', () => {
     });
   });
 });
+
+describe('USER CAN FOLLOW OTHER USERS', () => {
+  beforeEach(async () => {
+    await db.User.destroy({ cascade: true, truncate: true });
+  });
+
+  it('should be able to follow other users', async () => {
+    const firstUser = await createUser(await getUser());
+    const secondUser = await createUser(await getUser());
+    await createUser(await getUser());
+    const token = jwt.sign(firstUser, process.env.JWT_SECRET);
+
+    const res = await server()
+      .patch(`${users}/follow`)
+      .set('token', token)
+      .send({ followeeId: secondUser.id });
+    expect(res.status).to.equal(200);
+  });
+
+  it('should not be able to follow a user twice at the same time', async () => {
+    const firstUser = await createUser(await getUser());
+    const secondUser = await createUser(await getUser());
+    await baseRepository.create(db.Follower, {
+      followerId: firstUser.id,
+      followeeId: secondUser.id
+    });
+    await createUser(await getUser());
+    const token = jwt.sign(firstUser, process.env.JWT_SECRET);
+
+    const res = await server()
+      .patch(`${users}/follow`)
+      .set('token', token)
+      .send({ followeeId: secondUser.id });
+    expect(res.status).to.equal(400);
+    expect(res.body.message).to.equal(
+      `You were already following the user with id = ${secondUser.id}`
+    );
+  });
+
+  it('user should not be able to themselve', async () => {
+    const firstUser = await createUser(await getUser());
+    await createUser(await getUser());
+    const token = jwt.sign(firstUser, process.env.JWT_SECRET);
+
+    const res = await server()
+      .patch(`${users}/follow`)
+      .set('token', token)
+      .send({ followeeId: firstUser.id });
+    expect(res.status).to.equal(400);
+    expect(res.body.message).to.equal(`You cannot follow or unfollow yourself`);
```

Comment:
Assert that the follower count hasn't increased and followers doesn't include the user's id

Reply:
Thank you for your comment, Changes effected